crates/wallet/src/wallet/signer.rs

```diff
@@ -158,8 +158,12 @@ pub enum SignerError {
     NonStandardSighash,
     /// Invalid SIGHASH for the signing context in use
     InvalidSighash,
-    /// Error while computing the hash to sign
-    SighashError(sighash::Error),
+    /// Error while computing the hash to sign a P2WPKH input.
+    SighashP2wpkh(sighash::P2wpkhError),
+    /// Error while computing the hash to sign a Taproot input.
+    SighashTaproot(sighash::TaprootError),
+    /// Error while computing the hash, out of bounds access on the transaction inputs.
+    InputsIndexError(transaction::InputsIndexError),
```

Comment:
📌  Considering we already have a `InputIndexOutOfRange` error, I recommend consolidating these into one variant if possible.

Reply:
@ValuedMammal Oh, thanks! Yes, that's much better, I'll fix it.

Reply:
@ValuedMammal I did a deep-dive in rust-bitcoin, and now I'm wondering if we should have two variants, one for Psbt and another one for Transaction, as they have two in rust-bitcoin crate:
- bitcoin::psbt::IndexOutOfBoundsError
- bitcoin::transaction::InputsIndexError

Reply:
I updated it to be `TxInputsIndexError` instead, I also think for the current InputIndexOutOfRange it should be `PsbtInputIndexOutOfRange(psbt::IndexOutOfBoundsError)` instead, let me know what you think.

Reply:
A simple solution would be to make `SignerError::InputIndexOutOfRange` into a struct with fields for `index` and `length` (essentially reimplementing [`IndexOutOfBoundsError`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/transaction/struct.IndexOutOfBoundsError.html)) and then we have

```rust
impl From<transaction::InputsIndexError> for SignerError {}
```

similar to what you have here, but it shouldn't be a blocker to this PR. Accessing the outputs of a tx or psbt doesn't appear to be a concern in this module.